src/textual/app.py

```diff
@@ -939,6 +945,17 @@ async def on_event(self, event: events.Event) -> None:
             else:
                 # Forward the event to the view
                 await self.screen.forward_event(event)
+
+        elif isinstance(event, ModeReport):
```

Comment:
Probably shouldn't be an event. Makes more sense to be a `Message` which can be caught with `def handle_mode_report`.